src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -267,39 +292,44 @@ def _save_export(self, export):
         self.exports[self.rados_namespace].append(export)
         conf_block = export.to_export_block()
         self._write_raw_config(conf_block, "export-{}".format(export.export_id))
-        self._update_common_conf(export.export_id)
+        self._update_common_conf(export.cluster_id, export.export_id)
+
+    def create_export(self, fs_name, cluster_id, pseudo_path, read_only, path):
+        if not self.check_fs(fs_name):
+            return -errno.EINVAL,"", "Invalid CephFS name"
 
-    def create_export(self, export_type, fs_name, pseudo_path, read_only, path, cluster_id):
-        if export_type != 'cephfs':
-            return -errno.EINVAL,"", f"Invalid export type: {export_type}"
         #TODO Check if valid cluster
         if cluster_id not in self.exports:
             self.exports[cluster_id] = []
 
         self.rados_namespace = cluster_id
-        if not self.check_fs(fs_name) or self.check_pseudo_path(pseudo_path):
-            return -errno.EINVAL,"", "Invalid CephFS name or export already exists"
-
-        user_id, key = self._create_user_key(cluster_id)
-        if isinstance(user_id, int):
-            return user_id, "", key
         access_type = "RW"
         if read_only:
             access_type = "R"
 
-        ex_dict = {
-            'path': self.format_path(path),
-            'pseudo': self.format_path(pseudo_path),
-            'cluster_id': cluster_id,
-            'access_type': access_type,
-            'fsal': {"name": "CEPH", "user_id":cluster_id, "fs_name": fs_name, "sec_label_xattr": ""},
-            'clients': []
-            }
-
-        ex_id = self._gen_export_id()
-        export = Export.from_dict(ex_id, ex_dict)
-        export.fsal.cephx_key = key
-        self._save_export(export)
+        if not self._fetch_export(pseudo_path):
+            ex_id = self._gen_export_id()
+            user_id = f"{cluster_id}{ex_id}"
+            user_out, key = self._create_user_key(user_id, path, fs_name)
+
+            if isinstance(user_out, int):
+                return user_out, "", key
+
+            ex_dict = {
+                    'path': self.format_path(path),
+                    'pseudo': self.format_path(pseudo_path),
+                    'cluster_id': cluster_id,
+                    'access_type': access_type,
+                    'fsal': {"name": "CEPH", "user_id": user_id,
+                             "fs_name": fs_name, "sec_label_xattr": ""},
```

Comment:
Why do you need to set  'sec_label_xattr'  to an empty string? Currently, it never gets generated in the export block as it's an empty string.

Reply:
Based on the discussion [here](https://github.com/ceph/ceph/pull/25918#discussion_r251781446). We want it to be settable if the config options are parsed from a json file.